### ISSUE TITLE
Type error when using constructor initialization

### ISSUE THREAD

**Issue by MonoLogueChi** [hadashiA/VYaml#95]
eg

```cs
[YamlObject]
public partial class MyClass
{
  public MyClass(int a = 0, float b = 0f)
  {
    A = a;
    B = b;
  }
  public int A { get; set; }
  public float B { get; set; }
}
```

will generate

```cs
var __A__ = 0;
var __B__ = 0;
```

The type of B should be float, not int

**Comment by MonoLogueChi** [comment]
This method may solve the problem, but I'm not familiar with SourceGenerator and I'm not sure if this is correct.


https://github.com/hadashiA/VYaml/blob/b06c5ca126852eafdd669c0841f68db98d509dfd/VYaml.SourceGenerator/VYamlIncrementalSourceGenerator.cs#L442

```diff
- codeWriter.Append($"var __{memberMeta.Name}__ = ");
+ codeWriter.Append($"{memberMeta.FullTypeName} __{memberMeta.Name}__ = ");
```

**Comment by MonoLogueChi** [comment]
Enum also has the same problem and will be defined as int type

**Comment by hadashiA** [comment]
Thanks for the report.
Fixed in #102.
